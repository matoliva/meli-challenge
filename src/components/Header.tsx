import logo from '../assets/logo.png'

interface IProps {
  children?: React.ReactNode
}
export const Header = ({children}: IProps) => {
  return (
    <header className="header" data-testid="header">
      <img className="header__logo" src={logo} alt="logo" />
      {children}
    </header>
  )
}
