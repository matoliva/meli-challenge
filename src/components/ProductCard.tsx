import {IItems} from '../models/products'

interface IProps {
  item: IItems
}
export const ProductCard = ({item}: IProps) => {
  const {title, price, picture, condition, location} = item
  const currency = price.currency === 'ARS' ? '$' : 'uSd'

  return (
    <article className="product-card">
      <div className="product-card__image">
        <img src={picture} alt="product" />
      </div>
      <div className="product-card__description">
        <p className="product-card__description--price">
          {currency} {price.amount}
        </p>
        <p>{title}</p>
        <p>{condition}</p>
      </div>
      <div className="product-card__location">
        <p>{location}</p>
      </div>
    </article>
  )
}
