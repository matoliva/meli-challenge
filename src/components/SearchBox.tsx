import logo from '../assets/logo.png'

export const SearchBox = () => {
  return (
    <div className="search-box">
      <img className="search-box__logo" src={logo} alt="logo" />

      <form className="search-box__form">
        <input
          type="text"
          className="input-text"
          placeholder=" Nunca dejes de buscar"
        />
        <div className="search-box__form__icon pointer"></div>
      </form>
    </div>
  )
}
