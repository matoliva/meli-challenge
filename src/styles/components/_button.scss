.btn {
  background-color: #3483fa;
  color: $secondary;
  font-size: medium;
  width: 100%;
  border: none;
  padding: 0.5rem 1rem 0.5rem 1rem;
  border-radius: 4px;

  &:hover {
    background-color: #2968c8;
  }
}
