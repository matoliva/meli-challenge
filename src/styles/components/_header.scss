.header {
  display: flex;
  gap: 1rem;
  width: 100%;
  align-items: center;
  background-color: $primary;
  padding: 0 5rem 0 5rem;
  height: 3rem;

  @media screen and (max-width: $mobile) {
    padding: 0 1rem 0 1rem;
  }

  &__logo {
    max-width: 3rem;
  }
}
