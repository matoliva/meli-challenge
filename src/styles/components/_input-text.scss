.input-text {
  height: 2rem;
  width: 100%;
  border-radius: 2px;
  border: none;
  outline-color: $light-grey;
}
