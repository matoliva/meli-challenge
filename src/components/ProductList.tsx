import {IItem} from '../models/products'
import {ProductCard} from './ProductCard'

interface IProp {
  items: IItem[]
}

export const ProductList = ({items}: IProp) => {
  return (
    <section data-testid="product-list">
      {items.map(item => (
        <ProductCard key={item.id} item={item} />
      ))}
    </section>
  )
}
