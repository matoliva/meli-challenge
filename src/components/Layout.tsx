import {Header} from './Header'
import {SearchBox} from './SearchBox'

interface IProps {
  children: React.ReactNode
}
export const Layout = ({children}: IProps) => {
  const onSearchChange = (searchText: string): void => {
    console.log(searchText)
  }
  return (
    <div className="main">
      <Header>
        <SearchBox onSearchChange={onSearchChange} />
      </Header>
      <main className="section">{children}</main>
    </div>
  )
}
