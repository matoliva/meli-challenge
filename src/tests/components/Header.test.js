import {render} from '@testing-library/react'
import {Header} from '../../components/Header'

describe('Header', () => {
  it('renders correctly', () => {
    const {getByTestId} = render(<Header />)
    const header = getByTestId('header')
    expect(header).toMatchSnapshot()
  })
})
