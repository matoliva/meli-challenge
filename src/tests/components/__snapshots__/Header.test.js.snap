// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<header
  class="header"
  data-testid="header"
>
  <img
    alt="logo"
    class="header__logo"
    src="logo.png"
  />
</header>
`;
